src/main/java/uk/gov/ons/census/casesvc/service/ReceiptService.java

```diff
@@ -29,10 +32,18 @@ public void processReceipt(ResponseManagementEvent receiptEvent) {
     uacQidLink.setActive(false);
 
     Case caze = uacQidLink.getCaze();
-    caze.setReceiptReceived(true);
+
+    if (caze != null) {
+      caze.setReceiptReceived(true);
+      caseService.saveAndEmitCaseUpdatedEvent(caze);
+    } else {
+      log.with("qid", receiptPayload.getQuestionnaireId())
+          .with("tx_id", receiptEvent.getEvent().getTransactionId())
+          .with("channel", receiptEvent.getEvent().getChannel())
+          .warn("Receipt received for unaddressed UAC/QID pair not yet linked to a case");
+    }
```

Comment:
Is this meant to be in here? looks like stuff for decoupling response_received and questionnaire_linked

Reply:
yeah that's in production. We don't want to regress a fix we put in.